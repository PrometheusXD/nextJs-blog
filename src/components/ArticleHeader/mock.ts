import { ArticleHeaderProps } from '.';

export default {
  id: '603e56b939bd976b42d790a0',
  title: 'Vis nobis veritus maluisset id, nonumy adversarium',
  excerpt:
    'Quod oblique mei ne, cetero delicatissimi sea cu, cu est harum dicit verterem. Tota forensibus ut usu, luptatum dignissim persecuti vel ex. Mea nostrud instructior cu, et referrentur theophrastus eum. Pro in iuvaret senserit.',
  cover: {
    id: '603e5a2785540677309fd51b',
    alternativeText: '',
    url: 'https://res.cloudinary.com/dlizakp2a/image/upload/v1614699042/Origin_of_nature_by_Julian_Tomasini_dcfdd17cbe.jpg',
  },
  author: {
    id: '603e557239bd976b42d79094',
    displayName: 'Ot√°vio Miranda',
    slug: 'otavio-miranda',
  },
  categories: [
    {
      id: '603e55b539bd976b42d79097',
      displayName: 'JavaScript',
      slug: 'java-script',
    },
    {
      id: '603e55ef39bd976b42d7909a',
      displayName: 'React',
      slug: 'react',
    },
  ],
  createdAt: '2020-04-20',
} as ArticleHeaderProps;
