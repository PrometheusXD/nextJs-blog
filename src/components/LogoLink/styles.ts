import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  color: inherit;

  > img {
    width: 14rem;
    height: 14rem;
    border-radius: 50%;
  }
`;
