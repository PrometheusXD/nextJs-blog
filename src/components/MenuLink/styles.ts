import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display:block;
    padding-bottom: ${theme.spacings.small};
    font-size: 1.8;
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 300ms ease-in-out;

    :hover {
      border-right: 0.5rem solid ${theme.colors.secondary};
    }

    > a {
      color:${theme.colors.white};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      :hover {
        color: ${theme.colors.secondary};
      }
    }

  `}
`;
