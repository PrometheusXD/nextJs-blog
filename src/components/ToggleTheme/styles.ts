import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 2;
`;

export const Label = styled.label`
    position: relative;
    display: inline-block;
    width: 5rem;
    height: 2.5rem;
    line-height: 0;
    font-size: 0;
    overflow: hidden;
    color: rgba(0,0,0,0);
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + ${Slider} {
      background: ${theme.colors.secondary};
      border: 1px solid ${theme.colors.secondary};

      &:hover {
        border: 1px solid ${theme.colors.darkerGray};
      }
    }

    &:checked + ${Slider}:before {
      transform: translateX(1.9rem);
      background: ${theme.colors.primary};
    }

    &:hover + ${Slider} {
      border: 1px solid ${theme.colors.secondary};
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.darkerGray};
    border: 1px solid ${theme.colors.darkerGray};
    -webkit-transition: all 300ms ease-in-out;
    transition: all 300ms ease-in-out;
    border-radius: 2rem;

    &:before {
      position: absolute;
      content: "";
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.12rem;
      background: ${theme.colors.white};
      -webkit-transition: all 300ms ease-in-out;
      transition: all 300ms ease-in-out;
      border-radius: 50%;

    }
  `}
`;
