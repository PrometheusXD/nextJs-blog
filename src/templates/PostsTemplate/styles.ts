import styled, { css } from 'styled-components';

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.large} 0;
    display: flex;
    justify-content: center;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: none;
    padding: ${theme.spacings.small} ${theme.spacings.large};
    cursor: pointer;
    transition: all 300ms ease-in-out;


    &:disabled,
    &:disabled:hover {
      background: ${theme.colors.darkerGray};
      color: #FFFFFF;
    }

    &:hover {
      background: ${theme.colors.secondary};
      color: #FFFFFF;
    }
  `}
`;
