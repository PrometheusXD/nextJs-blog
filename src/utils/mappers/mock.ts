export default {
  data: {
    setting: {
      data: {
        id: '1',
        attributes: {
          blogName: 'Matheus Rodrigues',
          blogDescription:
            'ncdsbnci csnb cus cu ybsduc bsdcb isb icbs bcsc lsnc sklcnl nkncl s.',
          menuLink: [
            {
              id: '1',
              link: '#bla',
              text: 'bla',
              newTab: false,
            },
          ],
          logo: {
            data: {
              id: '3',
              attributes: {
                alternativeText: null,
                url: 'https://res.cloudinary.com/dcfobv3yq/image/upload/v1686858611/logo_a47d56dba6.svg',
              },
            },
          },
          text: 'cedcedcedcedcedce',
        },
      },
    },
    posts: {
      data: [
        {
          id: '2',
          attributes: {
            slug: 'creating-an-api-with-strapi',
            title: 'creating an api with strapi',
            excerpt:
              'creating a fucking api with the shit of strapi who everyone love but i hate',
            content:
              'lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ipsum ',
            createdAt: '2023-06-10T21:11:24.338Z',
            allowComments: false,
            cover: {
              data: null,
            },
            categories: {
              data: [
                {
                  id: '3',
                  attributes: {
                    displayName: 'Python',
                    slug: 'python',
                  },
                },
              ],
            },
            tags: {
              data: [
                {
                  id: '1',
                  attributes: {
                    displayName: 'useState',
                    slug: 'use-state',
                  },
                },
              ],
            },
            author: {
              data: {
                id: '2',
                attributes: {
                  displayName: 'lucas da silva',
                  slug: 'lucas-da-silva',
                },
              },
            },
          },
        },
        {
          id: '1',
          attributes: {
            slug: 'posnvdjknvj-jcnscn-sinc-ions-ncsn-jcnsmcmko',
            title: 'posnvdjknvj jcnscn sinc ions ncsn jcnsmcmko ',
            excerpt:
              'cndsicvdsiivcnsdui cnids ncisdb cisbic bsdib cisdbci bsicb sjkcb jksnc nscnsk klkwmcons jocnjias ncjsnc ljsnj cnsjnc sd jlcjsndioic nsdc sjkdcnsonc sn clsnc jsdjvbjajoisn kj njkn jsb jsb.\njcncjksjkcnsk.',
            content: '# jciodncodncod',
            createdAt: '2023-06-09T11:46:12.511Z',
            allowComments: true,
            cover: {
              data: null,
            },
            categories: {
              data: [
                {
                  id: '2',
                  attributes: {
                    displayName: 'React',
                    slug: 'react',
                  },
                },
                {
                  id: '1',
                  attributes: {
                    displayName: 'JavaScript',
                    slug: 'java-script',
                  },
                },
              ],
            },
            tags: {
              data: [
                {
                  id: '2',
                  attributes: {
                    displayName: 'useEffect',
                    slug: 'use-effect',
                  },
                },
                {
                  id: '1',
                  attributes: {
                    displayName: 'useState',
                    slug: 'use-state',
                  },
                },
              ],
            },
            author: {
              data: {
                id: '1',
                attributes: {
                  displayName: 'Matheus Rodrigues',
                  slug: 'matheus-rodrigues',
                },
              },
            },
          },
        },
      ],
    },
  },
};
