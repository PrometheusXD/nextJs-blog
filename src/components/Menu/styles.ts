import styled, { DefaultTheme, css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  menuVisible: boolean;
};

const wrapperChanger = (theme: DefaultTheme) =>
  css`
    left: -30rem;
    overflow-y: hidden;

    @media ${theme.media.lteMedium} {
    left: -32rem;
    }
`;

export const Wrapper = styled.div<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;

    ${!menuVisible && wrapperChanger(theme)}
  `}
`;

export const Nav = styled.nav`
  margin: auto;
  width: 100%;
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}
`;

const buttonChanger = (theme: DefaultTheme) =>
  css`
    left: 1rem;

    @media ${theme.media.lteMedium} {
    left: -0.5rem;
    }
`;

export const OpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    transition: all 300ms ease-in-out;

    &:hover {
    color: ${theme.colors.secondary};
    }

    ${!menuVisible && buttonChanger(theme)}
  `}
`;
