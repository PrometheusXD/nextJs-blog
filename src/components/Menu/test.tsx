import { fireEvent, screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { Menu, MenuProps } from '.';

import mock from './mock';

const props: MenuProps = mock;

describe('<Menu />', () => {
  it('should render button link', () => {
    renderTheme(<Menu {...props} links={undefined} />);

    const buttonLink = screen.getByRole('link', { name: 'Open or close menu' });
    const openMenuIcon = screen.getByLabelText('Open menu');

    expect(buttonLink).toBeInTheDocument();
    expect(openMenuIcon).toBeInTheDocument();
    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
    expect(screen.queryByRole('navigation')).not.toBeInTheDocument();
  });

  it('should open/close menu on button click', () => {
    const { container } = renderTheme(<Menu {...props} />);

    const buttonLink = screen.getByRole('link', { name: 'Open or close menu' });

    fireEvent.click(buttonLink);

    expect(screen.getByLabelText('Close menu')).toBeInTheDocument();
    expect(screen.queryByLabelText('Open menu')).not.toBeInTheDocument();
    expect(screen.getByRole('navigation')).toBeInTheDocument();
    expect(
      screen.getByRole('navigation').querySelectorAll('a:not([href="/"])'),
    ).toHaveLength(mock.links.length);
    expect(
      screen.getByRole('heading', { name: 'Otávio Miranda' }),
    ).toBeInTheDocument();
    expect(
      screen.getByRole('img', { name: 'Otávio Miranda' }),
    ).toBeInTheDocument();

    fireEvent.click(buttonLink);

    expect(screen.queryByLabelText('Close menu')).not.toBeInTheDocument();
    expect(screen.queryByRole('navigation')).not.toBeInTheDocument();
    expect(screen.getByLabelText('Open menu')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
