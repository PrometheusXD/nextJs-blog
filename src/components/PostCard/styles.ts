import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    box-shadow: 0 0 5px 0px ${theme.colors.darkerGray};

    &::after,
    &::before {
      content: "";
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      transition: transform .3s ease-in-out;
    }

    &::before {
      border-left: 2px solid ${theme.colors.primary};
      border-right: 2px solid ${theme.colors.primary};
      transform: scaleY(0);
    }

    &::after {
      border-top: 2px solid ${theme.colors.primary};
      border-bottom: 2px solid ${theme.colors.primary};
      transform: scaleX(0);
    }

    &:hover::before {
      transform: scaleY(1);
    }

    &:hover::after {
      transform: scaleX(1);
    }


    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    padding: calc(${theme.spacings.small} - 0.6rem);
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.darkText};
  `}
`;
