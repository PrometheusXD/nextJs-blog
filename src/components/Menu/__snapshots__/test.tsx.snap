// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should open/close menu on button click 1`] = `
.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: inherit;
}

.c7 > img {
  width: 14rem;
  height: 14rem;
  border-radius: 50%;
}

.c6 {
  color: #0A1128;
  line-height: 1.2;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c8 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #0A1128;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid #f231a5;
  color: #f231a5;
}

.c2 {
  background: #0A1128;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -30rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.0rem;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #0A1128;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: 1rem;
}

.c0:hover {
  color: #f231a5;
}

@media (max-width:768px) {
  .c2 {
    left: -32rem;
  }
}

@media (max-width:768px) {
  .c0 {
    left: -0.5rem;
  }
}

<div>
  <a
    aria-label="Open or close menu"
    class="c0"
    href="#"
    title="Open or close menu"
  >
    <svg
      aria-hidden="true"
      aria-label="Open menu"
      class="c1"
      fill="currentColor"
      focusable="false"
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
      />
    </svg>
  </a>
  <div
    aria-hidden="true"
    class="c2"
  >
    <nav
      class="c3"
    >
      <div
        class="c4"
      >
        <h1
          class="c5 c6"
        >
          <a
            href="/"
            target="_self"
          >
            <div
              class="c7"
            >
              <img
                alt="OtÃ¡vio Miranda"
                src="/assets/images/ana-banana.png"
              />
            </div>
          </a>
        </h1>
      </div>
      <a
        href="/home/1"
        target="_self"
      >
        <div
          class="c8"
        >
          My jobs
        </div>
      </a>
      <a
        href="/home/2"
        target="_self"
      >
        <div
          class="c8"
        >
          About me
        </div>
      </a>
      <a
        href="/home/3"
        target="_self"
      >
        <div
          class="c8"
        >
          Services
        </div>
      </a>
      <a
        href="http://localhost/home/4"
        target="_self"
      >
        <div
          class="c8"
        >
          External Link
        </div>
      </a>
      <a
        href="/home/5"
        target="_self"
      >
        <div
          class="c8"
        >
          JavaScript
        </div>
      </a>
      <a
        href="/home/6"
        target="_self"
      >
        <div
          class="c8"
        >
          Python
        </div>
      </a>
      <a
        href="/home/7"
        target="_self"
      >
        <div
          class="c8"
        >
          Consectetur
        </div>
      </a>
      <a
        href="/home/8"
        target="_self"
      >
        <div
          class="c8"
        >
          Text test
        </div>
      </a>
      <a
        href="/home/9"
        target="_self"
      >
        <div
          class="c8"
        >
          Pac-man
        </div>
      </a>
      <a
        href="/home/10"
        target="_self"
      >
        <div
          class="c8"
        >
          Star Wars
        </div>
      </a>
      <a
        href="/home/11"
        target="_self"
      >
        <div
          class="c8"
        >
          A menu
        </div>
      </a>
    </nav>
  </div>
</div>
`;
